operatorcert/__init__.py

```diff
@@ -311,3 +312,59 @@ def verify_pr_uniqueness(
             for duplicate in duplicate_prs:
                 logging.error(f"DUPLICATE: {duplicate}")
             raise RuntimeError("Multiple pull requests for one Operator Bundle")
+
+
+def download_artifacts(args, resource: str) -> Optional[str]:
+    """
+    Try to get the test-results or artifacts for given parameters from the Pyxis.
+    On success, store the results in files and return True, on failure- return False.
+    """
+
+    if resource not in ["test-results", "artifacts"]:
+        raise ValueError(
+            f"Preflight results are either test-results or artifacts. Got {resource} instead"
+        )
+
+    # If there are multiple results, we only want the most recent one- we enforce it by sorting by creation_date
+    # and getting only the first result.
+    test_results_url = urljoin(
+        args.pyxis_url,
+        f"v1/projects/certification/id/{args.cert_project_id}/{resource}?"
+        f"filter=certification_hash=='{args.certification_hash}';"
+        f"version=='{args.operator_package_version}';"
+        f"operator_package_name=='{args.operator_name}'"
+        f"&sort_by=creation_date[desc]&page_size=1",
```

Comment:
There can be more than just one artifact produced by a single pre-flight test. There should be at least a test log artifact + additional files produced by tests.

Reply:
So.. Do we want all of the artifacts for the same cert_project, cert_hash, version and operator package name?